### PR TITLE
Convert UDFs to SQLX and use Dataform for testing/deployment

### PR DESCRIPTION
**Pull request by danieldeleo**
Refactoring the bqutil UDFs and the unit testing framework to use SQLX syntax and features. Dataform will handle deployment and unit testing of UDFs.

### CODE REVIEW

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  fi
+}
+
+#######################################
+# description
+# Arguments:
+#   1
+#   2
+#   3
+#   4
```
Comment (ryanmcdowell) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:159]: Add documentation

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+#   JS_BUCKET
+#   PROJECT_ID
+#   SHORT_SHA
+# Arguments:
+#  None
+#######################################
+main() {
+  # For now, this build script assumes all BigQuery environments
+  # live in the same location which you specify below.
+  export BQ_LOCATION=US
```
Comment (ryanmcdowell) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:199]: Can we make this default to US but allow for overrides? Currently someone would have to modify the script to use a different location.

File: udfs/tests/dataform_testing_framework/includes/unit_test_utils.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+        var r = (Math.random() * 16) | 0,
+            v = c == "x" ? r : (r & 0x3) | 0x8;
+        return v.toString(16);
+    });
+}
+
+module.exports = {
+    generate_udf_test,
+};
```
Comment (ryanmcdowell) [inline udfs/tests/dataform_testing_framework/includes/unit_test_utils.js:97]: Trailing newline

File: udfs/CONTRIBUTING.md [hunk 4]
```diff
@@ -9,87 +9,94 @@ The following is a set of guidelines for contributing a UDF to this repository.
...
 
-1.  Create a Python virtual environment and activate it:
+   ```bash 
+   gcloud init
+   ```
 
-    *   `python3 -m venv venv`
-    *   `source venv/bin/activate`
-    *   `pip install -r tests/requirements.txt`
+1. Grant your Cloud Build service account the BigQuery Job User role
```
[the remark this answers is not in the document]
Comment (danieldeleo) [reply udfs/CONTRIBUTING.md:68]: good idea

File: udfs/community/ts_linear_interpolate.sqlx [hunk 5]
```diff
@@ -31,7 +32,7 @@
 * SELECT 
 *   *,
 *   COALESCE(coord.y,
-*     fn.ts_lin_interpolate(
+*     ts_lin_interpolate(
```
[the remark this answers is not in the document]
Comment (danieldeleo) [reply udfs/community/ts_linear_interpolate.sqlx:35]: good catch. this is a just the example provided that had a typo

**Comment by danieldeleo** [comment]
@ryanmcdowell thank you for all the feedback! All comments have been addressed

**Review by ryanmcdowell — APPROVED** [review]

**Pull request merged**